text.go

```diff
@@ -548,6 +548,20 @@ func (t *Text) ToPaths() ([]*Path, []color.RGBA) {
 	return paths, colors
 }
 
+// RenderDecoAsPath renders the text decorations using the provided RenderPath function.
+// Usually called with r.RenderPath as first argument
+func (t *Text) RenderDecoAsPath(render func(*Path, Style, Matrix), m Matrix) {
```

Comment:
I'd prefer passing a `Renderer` interface as that is the interface we're working with. Any renderer will have all methods implemented, no need to specialize smaller interfaces still. I guess we could name it `RenderDeco` as it is pretty obvious it is rendered as a path.

Reply:
:+1: 
I also added `TODO` comments for the compliance and overlapping text lines.

Since this PR does not change the behavior of the library, I would prefer having them addressed in another PR (probably when we need it - currently not my case :-)

Reply:
I agree on all points!